artiq/coredevice/phaser.py

```diff
@@ -549,6 +568,50 @@ def get_dac_temperature(self) -> TInt32:
         """
         return self.dac_read(0x06, div=257) >> 8
 
+    @kernel
+    def dac_sync(self):
+        """Trigger DAC synchronisation for both output channels.
+        If the DAC-NCO is enabled, this applies NCO frequency changes."""
+        config_1f = self.dac_read(0x1f)
+        delay(.1*ms)
+        self.dac_write(0x1f, config_1f | (1 << 1))
```

Comment:
As I explained, it's better to first clear and then set, and leave it set. Assuming the rising edge is what matters, this makes it idempotent. The way you do it will fail if someone sets it in the device_db.

Reply:
That may be tricky because of the temporal granularity of the commands in the control data stream.
But again to clarify: is it true that the rising edge of `sif_sync` does the synchronization (i.e. not level sensitive and not falling egde)? Then I would strongly prefer this method to do "clear-set" and not "set-clear" for robustness reasons.